### PR TITLE
align toolchain to operate more closely to latest changes in cmake toolchain

### PR DESCRIPTION
**Pull request by solvingj**
Changelog: Omit
Docs: Omit

Followup from https://github.com/conan-io/conan/pull/7738

Remove all conditional logic from generated toolchain file.  Refactor jinja2 templating. Refactor unit test methodology to make it easier to verify and update that output files look as expected.

### CODE REVIEW

File: conans/test/functional/toolchain/test_make.py [hunk 1]
```diff
@@ -159,11 +161,13 @@ def build(self):
 
         client.save(files_to_save, clean_first=True)
         client.run("install . hello/0.1@ %s %s" % (settings_str, options_str))
+        print(client.load("conan_toolchain.mak"))
```
Comment (memsharded) [inline conans/test/functional/toolchain/test_make.py]: Please remember to remove the prints()

File: conans/test/functional/toolchain/test_make.py [hunk 2]
```diff
@@ -159,11 +161,13 @@ def build(self):
...
 
         if target == "exe":
             client.run_command("make exe")
             client.run_command("./out/hello.bin")
             self.assertIn("Hello World {}!".format(build_type), client.out)
+            self.assertIn("Hello World {}!".format(build_type), client.out)
```
Comment (memsharded) [inline conans/test/functional/toolchain/test_make.py]: Duplicated check?

**Review by memsharded — APPROVED** [review]

File: conans/test/functional/toolchain/test_make.py (hunk 2 above)
Comment (solvingj) [reply conans/test/functional/toolchain/test_make.py]: fixed

File: conans/test/functional/toolchain/test_make.py (hunk 1 above)
Comment (solvingj) [reply conans/test/functional/toolchain/test_make.py]: fixed

**Pull request merged**